packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts

```diff
@@ -291,7 +291,9 @@ export class StripeHandler {
 
     await this.customerChanged(request, uid, email);
 
-    return { subscriptionId };
+    const sourceCountry = customer?.shipping?.address?.country || null;
```

Comment:
This is the main difference between [Ivo's original draft](https://paste.mozilla.org/QVeZPD0g) and this updated PR.

Previously, we were obtaining `sourceCountry` in one of two ways:
```js
const sourceCountry =
  this.stripeHelper.extractSourceCountryFromSubscription(subscription) ||
  customer?.shipping?.address?.country ||
  null;
```
AIUI, this first approach of using `extractSourceCountryFromSubscription` only works for Stripe CC customers. Unfortunately, for PayPal customers, we would have to look up their billing agreement details.

Since I originally wrote the ticket, I've learned that the Stripe Tax work puts the source country on the customer's shipping address, and this solution works for both Stripe and PayPal customers.